### PR TITLE
fix: allow commas in the template string in json body template

### PR DESCRIPTION
**Pull request by sumitaich1998**

### CODE REVIEW

File: src/main/java/com/gotocompany/depot/common/Template.java [hunk 1]
```diff
@@ -19,8 +19,9 @@ public Template(String template) throws InvalidTemplateException {
             throw new InvalidTemplateException("Template cannot be empty");
         }
         List<String> templateStrings = new ArrayList<>();
-        Splitter.on(",").omitEmptyStrings().split(template).forEach(s -> templateStrings.add(s.trim()));
-        this.templatePattern = templateStrings.get(0);
+
+        Splitter.onPattern("(?<!\\\\\",),(?!\\\")").omitEmptyStrings().split(template).forEach(s -> templateStrings.add(s.trim()));
```
Comment (rajuGT) [inline src/main/java/com/gotocompany/depot/common/Template.java:23]: The approach to escape comma \",\" looks good (ideal for programmers)

Can we consider it in simpler way /,/ by doing this we can make the escape literal smaller and also we can avoid the above two regular expressions? I'm not sure about the pitfalls of the suggested /,/ pattern, the intention is to have more user-friendly escape literal & not sure of performance impact of the above regular expressions which has look-behind & look-ahead regex. 
@sbchaos your thoughts?

File: src/test/java/com/gotocompany/depot/common/TemplateTest.java [hunk 2]
```diff
@@ -131,4 +131,64 @@ public void shouldGetTemplatePatternFromParameterizedString() throws InvalidTemp
...
+    }
+
+    @Test
+    public void shouldParseMultipleCommasEnclosedInDoubleQuotes() throws InvalidTemplateException {
+        Template template = new Template("eeeee\",\"2222\",\"eeeee\",\"2222");
+        assertEquals("eeeee,2222,eeeee,2222", template.getTemplateString());
+    }
+
+    @Test
+    public void shouldParseCommaEnclosedInDoubleQuotesWithArguments() throws InvalidTemplateException {
```
Comment (gdgagan696) [inline src/test/java/com/gotocompany/depot/common/TemplateTest.java:148]: You can use the Paratmeterised test here.

File: docs/sinks/http-sink.md [hunk 3]
```diff
@@ -63,6 +63,16 @@ Corresponding payloads-
...
 
+Note: If you want to add a comma inside the JSON String, then you need to enclose it with \", like \",\" 
+
+Examples Templates-
+
+* `SINK_HTTPV2_JSON_BODY_TEMPLATE="ww\",\"ee"`
+
+Corresponding payloads-
+
+* `"ww,ee""`
```
[the remark this answers is not in the document]
Comment (sumitaich1998) [reply docs/sinks/http-sink.md]: fixed

File: src/main/java/com/gotocompany/depot/common/Template.java (hunk 1 above)
Comment (sumitaich1998) [reply src/main/java/com/gotocompany/depot/common/Template.java:23]: ok let me try this

Comment (sumitaich1998) [reply src/main/java/com/gotocompany/depot/common/Template.java:23]: @rajuGT but the regular expression will still be required right? also performance wont be affected much , this template object is created only once during firehose starting

File: src/test/java/com/gotocompany/depot/common/TemplateTest.java (hunk 2 above)
Comment (sumitaich1998) [reply src/test/java/com/gotocompany/depot/common/TemplateTest.java]: done, converted to parameterized test, thanks

File: src/main/java/com/gotocompany/depot/common/Template.java (hunk 1 above)
Comment (sumitaich1998) [reply src/main/java/com/gotocompany/depot/common/Template.java]: done, ive changed the escape pattern to \,\

**Review by rajuGT — APPROVED** [review]

**Pull request merged**